### ISSUE TITLE
7.5 issue - Extra Gas Meter device appeared & rates not moving

### ISSUE THREAD

**Issue by JRascagneres** [BottlecapDave/HomeAssistant-OctopusEnergy#311]
PLEASE DO NOT DELETE AND FILL IN!

**Describe the bug**

I just updated the integration to the latest version that just became available and a random additional Gas Meter turned up:

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/3264971/7295cbd1-8310-4efc-9381-20eec2de0a13)

The first gas meter (second item in list) has nothing on it:

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/3264971/f7a044a6-a8d6-4b8b-bb13-09875a71b7bd)

The other is the functional one that I expect.

**To Reproduce**

Unknown, just updated the integration to the latest version. No other changes.

**Expected behavior**

Not have an extra device

**Home Assistant Version**

Home Assistant 2023.7.2

**Integration Version**

v7.5.0

**Fresh Install?**

Update integration

**Home Assistant Logs**

Will add later if required.

**Comment by JRascagneres** [comment]
It appears that this update also caused my agile rates to not update. I've rolled back to 7.4.5

**Comment by BottlecapDave** [comment]
Hello. It makes it a lot easier to track issues and fixes if each issue only contains a single problem.

The additional device must be a side effect of me fixing the identifiers, which has been using the wrong identifier and could have caused clashes. I'll investigate if there's a way of removing the old obsolete device automatically. This _shouldn't_ have any functional impact.

Secondly, nothing has changed around the retrieval of rate data in this release and it looks like you've had issues with rates before based on your other [discussion](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/309). Do you have any errors in your logs relating to the integration? If not, could you try [enabling debugging](https://www.home-assistant.io/docs/configuration/troubleshooting/#enabling-debug-logging) for the integration, wait a day and then disable and paste the results in here. Please make sure you look through the logs and remove anything you might deem identifiable before posting.

**Comment by JRascagneres** [comment]
Apologies. I'm going to close this. I ended up deleting the integration and re-creating it. I renamed some of the electricity entities because the newer ones were prepended with "octopus_energy" and my old ones weren't and I wanted to keep the sensor history. Following this it seems fine, possibly compounded updates had caused some weirdness in my config. I'll open another issue if I see anything else wrong and will be sure to get a days worth of debug first. Thanks for all your help!
